force-app/src/classes/MethodSpy.cls

```diff
@@ -102,7 +102,27 @@ public class MethodSpy {
     this.exceptionToThrow = exceptionToThrow;
```

Comment:
**question:** what about empty param ?

Should we add an API for that instead of using `Params.empty()`
ie: void whenCalledWith();

It is equivalent to global returns...